### PR TITLE
Remove duplicates in Search Api fields

### PR DESCRIPTION
**Pull request by francistagbo**
Remove duplicates in Search Api fields:
- aggregate
- withField
- sortBy

### CODE REVIEW

**Review by nitzanj — CHANGES_REQUESTED** [review]
The code looks good but without the appropriate test cases we can't be sure it works
Please add tests that specify duplicate fields and see that the query is cleaning those out.

**Review by const-cloudinary — CHANGES_REQUESTED** [review]
Where is the check for duplicates?

**Comment by francistagbo** [comment]
Hi @const-cloudinary,

It is in the first commit, the Search.java class

**Comment by adimiz1** [comment]
Hi @francistagbo 
@const-cloudinary meant where are the duplicate checks in the tests added to this push
I just added them.

**Comment by francistagbo** [comment]
Okay, cool! Please let me know if you need anything from my side :)

File: cloudinary-core/src/main/java/com/cloudinary/Search.java [hunk 1]
```diff
@@ -24,6 +24,14 @@
         this.withFieldParam = new ArrayList<String>();
     }
 
+    public ArrayList<String> getAggregateParam() {
```
Comment (const-cloudinary) [inline cloudinary-core/src/main/java/com/cloudinary/Search.java:27]: Do you need these public getters?

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: cloudinary-core/src/main/java/com/cloudinary/Search.java (hunk 1 above)
Comment (adimiz1) [reply cloudinary-core/src/main/java/com/cloudinary/Search.java:27]: Removed

File: cloudinary-core/src/main/java/com/cloudinary/Search.java [hunk 2]
```diff
@@ -40,18 +48,28 @@ public Search nextCursor(String value) {
     }
 
     public Search aggregate(String field) {
-        aggregateParam.add(field);
+        if (!aggregateParam.contains(field)) {
```
[the remark this answers is not in the document]
Comment (const-cloudinary) [reply cloudinary-core/src/main/java/com/cloudinary/Search.java:51]: never mind, java 6

**Review by const-cloudinary — APPROVED** [review]
